### DISCUSSION TITLE
What's the mathematical formula of 'deviatoric strain' in Anura 3D?

### DISCUSSION THREAD

**Discussion by LiuTingy** [Anura3D/Anura3D_OpenSource#62] [Q&A] [answered]
Hello everyone:
    I am using Anura3D 2022 version and confused about the definition of 'deviatoric strain' in the output result. I couldn't find the explanation of it from the manual files. The mathematic formula of 'deviatoric strain' from other references is as the picture illustrates below:
![T)2G5ESO6F_NJ`4~_6XPX~I](https://user-images.githubusercontent.com/109811235/216760282-e8cfad59-6b14-49b3-80a1-077e9710b531.png)
Is that right? Hope for your anwsers.

**Comment by murphy-john** [comment, marked as answer]
What "output result" do you mean? The VTK files, the GiD output files, or other? 

In most cases, including the places the code I could find that outputs "deviatoric_strain_solid", the deviatoric strain is calculated in the function "getEpsq". The "getEpsq" function is in the "MatrixMath" module. The function is on _line 92_ of that file (Note: the line number may vary slightly in your version of the code). You can follow how deviatoric strain is calculated there.
